pkg/kubelet/container/helpers_test.go

```diff
@@ -414,6 +414,11 @@ func TestShouldContainerBeRestarted(t *testing.T) {
 		Name:      pod.Name,
 		Namespace: pod.Namespace,
 		ContainerStatuses: []*ContainerStatus{
+			{
```

Comment:
Sorry, but why does the order matter here? How come this change alters the test? (maybe I just need more coffee...)

Reply:
The previous test relies on the container with ContainerStateRunning to be first in the slice.

The logic in this PR scans the slice to find the container with ContainerStateRunning, hence it is not dependent on order of container states.